paths/pythonParser.py

```diff
@@ -0,0 +1,190 @@
+import sys
+from io import StringIO
+import time
+import random
+import os
+
+
+def main():
+    fileName = "paths\\" + sys.argv[1]
+    os.chdir("..")
+    path = os.getcwd()
+    if fileName[len(fileName) - 4:] != ".txt":
+        fileName += ".txt"
+    i = 0
+    velocities = []
+    rotations = []
+    frontLeftEncoderPositions = []
+    frontRightEncoderPositions = []
+    backLeftEncoderPositions = []
+    backRightEncoderPositions = []
+
+    bad_chars = ['ï»¿', 'Ã¯Â»Â¿']
+    splashText = ["The Robot Revolution Has Begun", "The Router Has Become Sentient", "Enter The Matrix",
+                  "Sky Net is Here", "CIMS Is Better Than POE", "It's DE Friday", "Auto Works!", "2 + 3 is 5",
+                  "Interpolating is hard", "The Meaning of Life is 42", "Don't Use Do While loops",
+                  "Edd, I Hardly Know Her",
+                  "How many programmers does it take to change a light bulb?\nNone – It’s a hardware problem",
+                  "Why do programmers always mix up Halloween and Christmas?\nBecause Oct 31 equals Dec 25"]
+    copy_right = ["UR MOM", "UR DAD", "HOWARD THE DUCK", "THE FIRST ORDER", "THANOS", "THE COSMIC GOAT",
+                  "THE SLACKER MENTOR"]
+
+    with open(fileName) as f, open(fileName[:len(fileName) - 4] + 'GraphData.txt', "w") as out, open(
+            fileName[:len(fileName) - 4] + 'DiscardedData.txt', "w") as trash:
+        sadness = random.randint(1, 1000000)
+        text = random.randint(0, len(splashText) - 1)
+        lol = random.randint(0, len(splashText) - 1)
+        e1 = random.randint(0, len(splashText) - 1)
+        e2 = random.randint(0, len(splashText) - 1)
+        ree = random.randint(0, len(copy_right) - 1)
+        year = random.randint(1945, 3099)
+        f_str = f.read()
+        f_str.encode('ascii', 'replace')
+        for a in bad_chars:
+            trash.write(a)
+            f_str = f_str.replace(a, '')
+        g = StringIO(f_str)
+        print(" Python Parser is about to Parse in Python")
+        time.sleep(1.5)
+        print(" Parsing in")
+        time.sleep(0.75)
+        print("  3")
+        time.sleep(0.5)
+        print("  2")
+        time.sleep(0.5)
+        print("  1")
+        time.sleep(0.5)
+        print(" Python Parser Parsed in Python Succsesfully")
+        time.sleep(0.25)
```

Comment:
I sense some easy speedup opportunities that we should take advantage of before merging

Reply:
it adds to the experience